micall/monitor/qai_helper.py

```diff
@@ -15,9 +15,12 @@ def login(self, qai_path, qai_user, password):
         """
         self.qai_path = qai_path
 
-        response = self.post(qai_path + "/account/login",
-                             data={'user_login': qai_user,
-                                   'user_password': password})
+        try:
+            response = self.post(qai_path + "/account/login",
+                                data={'user_login': qai_user,
+                                    'user_password': password})
+        except Exception as e:
+            raise RuntimeError('Could not log in to QAI') from None
```

Comment:
Why hide the original exception? Consider these options:

1. Don't catch the exception here. Just let the original exception bubble up to the upload loop and get logged there.
2. If the original exception doesn't have enough information to make it clear that it happened while trying to log in to QAI, just remove the `from None`. That will log both the original exception and the fact that it was trying to log in to QAI. You can be even more explicit about the link between the two exceptions by using `from e`.
3. If there is something in the original exception that was harmful or confusing to log, add a comment to explain why you added the `from None`.

In option 2 or 3, you should probably be more specific about the exception that you're catching. At least `RequestException`, and maybe `ConnectionError`.

Reply:
The traceback of the exception that is caught at this stage is quite verbose and most of it isn't really helpful, hence why I opted for this method (especially if it's part of a loop structure). I think the best option would be number 3 here, though I agree I could make the exception more specific. 

To resolve this I will:

1. Add a comment explaining `from None` was added to reduce the log message size which involves several nested exceptions being handled
2. Replace the `Exception` with `ConnectionError`

Reply:
I was suggesting that you *catch* a more specific exception class. Throwing the `RuntimeError` is fine. The rest of your plan sounds good to me.

Reply:
I see, I've replaced it with `RequestException`